Classes/Transfer/ContentReleaseCleaner.php

```diff
@@ -74,7 +81,10 @@ public function removeOldReleases(RedisInstanceIdentifier $redisInstanceIdentifi
             } elseif ($id->equals($contentReleaseIdentifierOfUpcomingRelease)) {
                 $contentReleaseLogger->info('- ' . $id->getIdentifier() . ' (UPCOMING)');
             } else {
-                $i++;
+                // on primary: only add to counter if status of release is "success" and release has no errors
+                if (!$redisInstanceIdentifier->isPrimary() || ($this->redisContentReleaseService->fetchMetadataForContentRelease($id)->getStatus()->getStatus() === NodeRenderingCompletionStatus::success()->getStatus() && count($this->redisRenderingErrorManager->getRenderingErrors($id)) === 0)) {
```

Comment:
Rename: $i -> $healthyReleaseCounter
```
if (!isPrimary) {
  // we are on a PRODUCTION Content Store (in case of a replicated setup) - there are no errors transferred there;
// and we want to NEVER run into a memory problem (that's why the cleanup always should run)
  $i++;
} else {
  // we are on the primary Content Store (the one which the pipeline uses as scratch space).
  // In case of errors, we want to keep more "good" content releases (without errors), so that we can
  // more easily switch back to an older release.
  // -> we accept potential Redis out of memory errors in this case.
  if (status == success && noRenderingErrors) {
    $i++;
  }
}
```